src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1267,7 +1292,12 @@ public void addExperience(final double xp)
                 return;
             }
 
-            double localXp = xp * skillModifier / EXP_DIVIDER;
+            double currenSkillModifier = skillModifier;
+            if (currenSkillModifier == 0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Equality tests should not be made with floating point values. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1244)